src/builtins/Transformers.jl

```diff
@@ -342,15 +342,61 @@ called.
 mutable struct UnivariateTimeTypeToContinuous <: Unsupervised
     zero_time::Union{Nothing, TimeType}
     step::Period
-    function UnivariateTimeTypeToContinuous(;
-        zero_time=nothing, step=Dates.Hour(24))
-        return new(zero_time, step)
+end
+
+function UnivariateTimeTypeToContinuous(;
+    zero_time=nothing, step=Dates.Hour(24))
+    model = UnivariateTimeTypeToContinuous(zero_time, step)
+    MLJBase.clean!(model)
+    return return model
+end
+
+function MLJBase.clean!(model::UnivariateTimeTypeToContinuous)
+    # Step must be able to be added to zero_time if provided.
+    if model.zero_time !== nothing
+        try
+            tmp = model.zero_time + model.step
+        catch err
+            if err isa MethodError
+                # Cannot add time parts to dates nor date parts to times.
+                # If a mismatch is encountered. Conversion from date parts to time parts
+                # is possible, but not from time parts to date parts because we cannot
+                # represent fractional date parts.
+                if model.zero_time isa Dates.Date && model.step isa Dates.TimePeriod
+                    # Convert zero_time to a DateTime to resolve conflict.
+                    @warn "Cannot add TimePeriod step to Date zero_time. Converting zero_time to DateTime."
+                    model.zero_time = convert(DateTime, model.zero_time)
+                elseif model.zero_time isa Dates.Time && model.step isa Dates.DatePeriod
+                    # Convert step to Hour if possible. This will fail for
+                    # isa(step, Month)
+                    @warn "Cannot add DatePeriod step to Time zero_time. Converting step to Hour."
+                    model.step = convert(Hour, model.step)
+                else
+                    # Unable to resolve, rethrow original error.
+                    throw(err)
+                end
+            else
+                throw(err)
+            end
+        end
     end
```

Comment:
Small misunderstanding here: `clean!` does not actually issue the warnings, just appends them to it's `message` return value (initialised to "" at the beginning). Please see the example at  https://alan-turing-institute.github.io/MLJ.jl/dev/adding_models_for_general_use/#New-model-type-declarations-and-optional-clean!-method-1 .

Reply:
Finicky point: When referring to fields in the warnings, put in quotes, as is in "The value of `lambda` is not valid".

Reply:
Fixed the `clean!` method semantics, but forgot the backticks. If I get time I'll revisit, but not sure when that will be.